WARNING:pygls.protocol:Cancel notification for unknown message id "122"
WARNING:pygls.protocol:Cancel notification for unknown message id "123"
WARNING:pygls.protocol:Cancel notification for unknown message id "131"
Exception in thread Thread-62:
Traceback (most recent call last):
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi_language_server/jedi_utils.py", line 289, in lsp_python_diagnostic
    compile(source, uri, "exec", PyCF_ONLY_AST)
  File "file:///home/nxtperfect/Documents/Programming/Python/BiggerProjects/Rotation/main.py", line 76
    class Mouse_movement:
IndentationError: expected an indented block after function definition on line 72

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.10/threading.py", line 1378, in run
    self.function(*self.args, **self.kwargs)
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi_language_server/jedi_utils.py", line 61, in run
    return func(*args, **kwargs)
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi_language_server/server.py", line 651, in _publish_diagnostics
    diagnostic = jedi_utils.lsp_python_diagnostic(uri, doc.source)
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi_language_server/jedi_utils.py", line 310, in lsp_python_diagnostic
    end=Position(line=until_line, character=until_column),
  File "<attrs generated init lsprotocol.types.Position>", line 7, in __init__
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/lsprotocol/validators.py", line 45, in uinteger_validator
    raise ValueError(
ValueError: Position.character should be in range [0:2147483647], but was -1.
WARNING:pygls.protocol:Cancel notification for unknown message id "384"
WARNING:pygls.protocol:Cancel notification for unknown message id "482"
WARNING:pygls.protocol:Cancel notification for unknown message id "732"
WARNING:pygls.protocol:Cancel notification for unknown message id "829"
WARNING:pygls.protocol:Cancel notification for unknown message id "1009"
WARNING:pygls.protocol:Cancel notification for unknown message id "1015"
WARNING:pygls.protocol:Cancel notification for unknown message id "1109"
WARNING:pygls.protocol:Cancel notification for unknown message id "1110"
WARNING:pygls.protocol:Cancel notification for unknown message id "1116"
WARNING:pygls.protocol:Cancel notification for unknown message id "1147"
WARNING:pygls.protocol:Cancel notification for unknown message id "1195"
WARNING:pygls.protocol:Cancel notification for unknown message id "1277"
WARNING:pygls.protocol:Cancel notification for unknown message id "1291"
WARNING:pygls.protocol:Cancel notification for unknown message id "1301"
WARNING:pygls.protocol:Cancel notification for unknown message id "1354"
WARNING:pygls.protocol:Cancel notification for unknown message id "1362"
WARNING:pygls.protocol:Cancel notification for unknown message id "1375"
WARNING:pygls.protocol:Cancel notification for unknown message id "1412"
WARNING:pygls.protocol:Cancel notification for unknown message id "1456"
WARNING:pygls.protocol:Cancel notification for unknown message id "1506"
WARNING:pygls.protocol:Cancel notification for unknown message id "1513"
WARNING:pygls.protocol:Cancel notification for unknown message id "1563"
WARNING:pygls.protocol:Cancel notification for unknown message id "1570"
WARNING:pygls.protocol:Cancel notification for unknown message id "1573"
WARNING:pygls.protocol:Cancel notification for unknown message id "1571"
WARNING:pygls.protocol:Cancel notification for unknown message id "1574"
WARNING:pygls.protocol:Cancel notification for unknown message id "1577"
WARNING:pygls.protocol:Cancel notification for unknown message id "1578"
WARNING:pygls.protocol:Cancel notification for unknown message id "1576"
WARNING:pygls.protocol:Cancel notification for unknown message id "1579"
WARNING:pygls.protocol:Cancel notification for unknown message id "1575"
WARNING:pygls.protocol:Cancel notification for unknown message id "1580"
WARNING:pygls.protocol:Cancel notification for unknown message id "1583"
WARNING:pygls.protocol:Cancel notification for unknown message id "1640"
WARNING:pygls.protocol:Cancel notification for unknown message id "1669"
WARNING:pygls.protocol:Cancel notification for unknown message id "1852"
WARNING:pygls.protocol:Cancel notification for unknown message id "1931"
WARNING:pygls.protocol:Cancel notification for unknown message id "1932"
WARNING:pygls.protocol:Cancel notification for unknown message id "2141"
WARNING:pygls.protocol:Cancel notification for unknown message id "2143"
WARNING:pygls.protocol:Cancel notification for unknown message id "2145"
WARNING:pygls.protocol:Cancel notification for unknown message id "2147"
WARNING:pygls.protocol:Cancel notification for unknown message id "2220"
WARNING:pygls.protocol:Cancel notification for unknown message id "2332"
WARNING:pygls.protocol:Cancel notification for unknown message id "2361"
WARNING:pygls.protocol:Cancel notification for unknown message id "2362"
WARNING:pygls.protocol:Cancel notification for unknown message id "2363"
WARNING:pygls.protocol:Cancel notification for unknown message id "2368"
WARNING:pygls.protocol:Cancel notification for unknown message id "2370"
WARNING:pygls.protocol:Cancel notification for unknown message id "2372"
WARNING:pygls.protocol:Cancel notification for unknown message id "2373"
WARNING:pygls.protocol:Cancel notification for unknown message id "2369"
WARNING:pygls.protocol:Cancel notification for unknown message id "2374"
WARNING:pygls.protocol:Cancel notification for unknown message id "2375"
WARNING:pygls.protocol:Cancel notification for unknown message id "2376"
WARNING:pygls.protocol:Cancel notification for unknown message id "2377"
WARNING:pygls.protocol:Cancel notification for unknown message id "2379"
WARNING:pygls.protocol:Cancel notification for unknown message id "2378"
WARNING:pygls.protocol:Cancel notification for unknown message id "2380"
WARNING:pygls.protocol:Cancel notification for unknown message id "2382"
WARNING:pygls.protocol:Cancel notification for unknown message id "2415"
WARNING:pygls.protocol:Cancel notification for unknown message id "2568"
WARNING:pygls.protocol:Cancel notification for unknown message id "2673"
WARNING:pygls.protocol:Cancel notification for unknown message id "2773"
WARNING:pygls.protocol:Cancel notification for unknown message id "2774"
WARNING:pygls.protocol:Cancel notification for unknown message id "2795"
ERROR:pygls.protocol:Failed to handle request 2819 textDocument/codeAction CodeActionParams(text_document=TextDocumentIdentifier(uri='file:///home/nxtperfect/Documents/Programming/Python/BiggerProjects/Rotation/main.py'), range=82:29-82:29, context=CodeActionContext(diagnostics=[], only=None, trigger_kind=<CodeActionTriggerKind.Automatic: 2>), work_done_token=None, partial_result_token=None)
Traceback (most recent call last):
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/pygls/protocol.py", line 337, in _handle_request
    self._execute_request(msg_id, handler, params)
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/pygls/protocol.py", line 261, in _execute_request
    self._send_response(msg_id, handler(params))
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi_language_server/server.py", line 556, in code_action
    inline_refactoring = jedi_script.inline(*jedi_lines)
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi/api/__init__.py", line 684, in inline
    names = [d._name for d in self.get_references(line, column, include_builtins=True)]
  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi/api/helpers.py", line 484, in wrapper
    raise ValueError('`column` parameter (%d) is not in a valid range '
ValueError: `column` parameter (29) is not in a valid range (0-28) for line 83 ('        self.keybind_button,\n').
[Error - 6:02:55 PM] Request textDocument/codeAction failed.
  Message: ValueError: `column` parameter (29) is not in a valid range (0-28) for line 83 ('        self.keybind_button,\n').
  Code: -32602 
{'traceback': ['  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/pygls/protocol.py", line 337, in _handle_request\n    self._execute_request(msg_id, handler, params)\n', '  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/pygls/protocol.py", line 261, in _execute_request\n    self._send_response(msg_id, handler(params))\n', '  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi_language_server/server.py", line 556, in code_action\n    inline_refactoring = jedi_script.inline(*jedi_lines)\n', '  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi/api/__init__.py", line 684, in inline\n    names = [d._name for d in self.get_references(line, column, include_builtins=True)]\n', '  File "/home/nxtperfect/.vscode-oss/extensions/ms-python.python-2023.4.1-universal/pythonFiles/lib/jedilsp/jedi/api/helpers.py", line 484, in wrapper\n    raise ValueError(\'`column` parameter (%d) is not in a valid range \'\n']}
WARNING:pygls.protocol:Cancel notification for unknown message id "2819"
WARNING:pygls.protocol:Cancel notification for unknown message id "2822"
WARNING:pygls.protocol:Cancel notification for unknown message id "2884"
WARNING:pygls.protocol:Cancel notification for unknown message id "2885"
WARNING:pygls.protocol:Cancel notification for unknown message id "2886"
WARNING:pygls.protocol:Cancel notification for unknown message id "2887"
WARNING:pygls.protocol:Cancel notification for unknown message id "2888"
WARNING:pygls.protocol:Cancel notification for unknown message id "2889"
WARNING:pygls.protocol:Cancel notification for unknown message id "2916"
WARNING:pygls.protocol:Cancel notification for unknown message id "2917"
WARNING:pygls.protocol:Cancel notification for unknown message id "2952"
WARNING:pygls.protocol:Cancel notification for unknown message id "2991"
WARNING:pygls.protocol:Cancel notification for unknown message id "3048"
WARNING:pygls.protocol:Cancel notification for unknown message id "3085"
WARNING:pygls.protocol:Cancel notification for unknown message id "3396"
WARNING:pygls.protocol:Cancel notification for unknown message id "3525"
WARNING:pygls.protocol:Cancel notification for unknown message id "3529"
WARNING:pygls.protocol:Cancel notification for unknown message id "3545"
WARNING:pygls.protocol:Cancel notification for unknown message id "3552"
WARNING:pygls.protocol:Cancel notification for unknown message id "3555"
WARNING:pygls.protocol:Cancel notification for unknown message id "3569"
WARNING:pygls.protocol:Cancel notification for unknown message id "3583"
WARNING:pygls.protocol:Cancel notification for unknown message id "3586"
WARNING:pygls.protocol:Cancel notification for unknown message id "3588"
WARNING:pygls.protocol:Cancel notification for unknown message id "3645"
WARNING:pygls.protocol:Cancel notification for unknown message id "3660"
WARNING:pygls.protocol:Cancel notification for unknown message id "3661"
WARNING:pygls.protocol:Cancel notification for unknown message id "3662"
WARNING:pygls.protocol:Cancel notification for unknown message id "3741"
WARNING:pygls.protocol:Cancel notification for unknown message id "3862"
WARNING:pygls.protocol:Cancel notification for unknown message id "3864"
WARNING:pygls.protocol:Cancel notification for unknown message id "3870"
WARNING:pygls.protocol:Cancel notification for unknown message id "3908"
WARNING:pygls.protocol:Cancel notification for unknown message id "4005"
WARNING:pygls.protocol:Cancel notification for unknown message id "4334"
WARNING:pygls.protocol:Cancel notification for unknown message id "4338"
WARNING:pygls.protocol:Cancel notification for unknown message id "4339"
WARNING:pygls.protocol:Cancel notification for unknown message id "4400"
WARNING:pygls.protocol:Cancel notification for unknown message id "4498"
WARNING:pygls.protocol:Cancel notification for unknown message id "4504"
WARNING:pygls.protocol:Cancel notification for unknown message id "4544"
WARNING:pygls.protocol:Cancel notification for unknown message id "4545"
WARNING:pygls.protocol:Cancel notification for unknown message id "4547"
WARNING:pygls.protocol:Cancel notification for unknown message id "4615"
WARNING:pygls.protocol:Cancel notification for unknown message id "4800"
WARNING:pygls.protocol:Cancel notification for unknown message id "4799"
WARNING:pygls.protocol:Cancel notification for unknown message id "4801"
WARNING:pygls.protocol:Cancel notification for unknown message id "4803"
WARNING:pygls.protocol:Cancel notification for unknown message id "4804"
WARNING:pygls.protocol:Cancel notification for unknown message id "4805"
WARNING:pygls.protocol:Cancel notification for unknown message id "4806"
WARNING:pygls.protocol:Cancel notification for unknown message id "4802"
WARNING:pygls.protocol:Cancel notification for unknown message id "4817"
WARNING:pygls.protocol:Cancel notification for unknown message id "4818"
WARNING:pygls.protocol:Cancel notification for unknown message id "4956"
WARNING:pygls.protocol:Cancel notification for unknown message id "4958"
WARNING:pygls.protocol:Cancel notification for unknown message id "5068"
WARNING:pygls.protocol:Cancel notification for unknown message id "5102"
WARNING:pygls.protocol:Cancel notification for unknown message id "5105"
WARNING:pygls.protocol:Cancel notification for unknown message id "5108"
WARNING:pygls.protocol:Cancel notification for unknown message id "5123"
WARNING:pygls.protocol:Cancel notification for unknown message id "5137"
WARNING:pygls.protocol:Cancel notification for unknown message id "5257"
WARNING:pygls.protocol:Cancel notification for unknown message id "5258"
WARNING:pygls.protocol:Cancel notification for unknown message id "5259"
WARNING:pygls.protocol:Cancel notification for unknown message id "5310"
WARNING:pygls.protocol:Cancel notification for unknown message id "5327"
WARNING:pygls.protocol:Cancel notification for unknown message id "5349"
WARNING:pygls.protocol:Cancel notification for unknown message id "5434"
WARNING:pygls.protocol:Cancel notification for unknown message id "5440"
WARNING:pygls.protocol:Cancel notification for unknown message id "5446"
WARNING:pygls.protocol:Cancel notification for unknown message id "5461"
WARNING:pygls.protocol:Cancel notification for unknown message id "5475"
WARNING:pygls.protocol:Cancel notification for unknown message id "5507"
WARNING:pygls.protocol:Cancel notification for unknown message id "5531"
WARNING:pygls.protocol:Cancel notification for unknown message id "5590"
WARNING:pygls.protocol:Cancel notification for unknown message id "5591"
WARNING:pygls.protocol:Cancel notification for unknown message id "5593"
WARNING:pygls.protocol:Cancel notification for unknown message id "5611"
WARNING:pygls.protocol:Cancel notification for unknown message id "5689"
WARNING:pygls.protocol:Cancel notification for unknown message id "5807"
WARNING:pygls.protocol:Cancel notification for unknown message id "5838"
WARNING:pygls.protocol:Cancel notification for unknown message id "6003"
WARNING:pygls.protocol:Cancel notification for unknown message id "6034"
WARNING:pygls.protocol:Cancel notification for unknown message id "6117"
WARNING:pygls.protocol:Cancel notification for unknown message id "6125"
WARNING:pygls.protocol:Cancel notification for unknown message id "6129"
WARNING:pygls.protocol:Cancel notification for unknown message id "6144"
WARNING:pygls.protocol:Cancel notification for unknown message id "6218"
WARNING:pygls.protocol:Cancel notification for unknown message id "6225"
WARNING:pygls.protocol:Cancel notification for unknown message id "6381"
WARNING:pygls.protocol:Cancel notification for unknown message id "6396"
WARNING:pygls.protocol:Cancel notification for unknown message id "6412"
WARNING:pygls.protocol:Cancel notification for unknown message id "6414"
WARNING:pygls.protocol:Cancel notification for unknown message id "6413"
WARNING:pygls.protocol:Cancel notification for unknown message id "6416"
WARNING:pygls.protocol:Cancel notification for unknown message id "6415"
WARNING:pygls.protocol:Cancel notification for unknown message id "6474"
WARNING:pygls.protocol:Cancel notification for unknown message id "6475"
WARNING:pygls.protocol:Cancel notification for unknown message id "6479"
WARNING:pygls.protocol:Cancel notification for unknown message id "6531"
WARNING:pygls.protocol:Cancel notification for unknown message id "6917"
WARNING:pygls.protocol:Cancel notification for unknown message id "6950"
WARNING:pygls.protocol:Cancel notification for unknown message id "7016"
WARNING:pygls.protocol:Cancel notification for unknown message id "7018"
